### PR TITLE
fix(behavior_path_planner): lane change ego and target polygon pose

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
Signed-off-by: Muhammad Zulfaqar Azmi <zulfaqar.azmi@tier4.jp>

## Description

The current lane change ego polygon's pose is not same with ego actual pose.
Furthermore, when due to floating error when computing projected point on polygon segment, the position of object and ego keeps on changing when both is not moving.

The PR aims to solve this issue.

Before PR (ego's polygon position is wrong, and `position` details in object marker is alternating constantly.

https://user-images.githubusercontent.com/93502286/188429190-56459bc4-fe6d-4d43-bb07-73eaf724cae0.mp4

After PR (ego's polygon position is correct, and `position` details in object marker is stable.

https://user-images.githubusercontent.com/93502286/188428550-a57759a8-3124-40d3-91bd-638a60bac943.mp4

## Tests performed

Use PSIM to simulate lane change and turn on debug marker.

## Notes for reviewers

This PR also include refactoring.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/lane_change/util.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ std::vector<LaneChangePath> getLaneChangePaths(
       minimum_lane_change_prepare_distance);
     const double lane_changing_distance = getDistanceWhenDecelerate(
       lane_change_prepare_velocity, acceleration, lane_changing_duration,
-      minimum_lane_change_length + buffer);
+      minimum_lane_change_length);
```
Comment (shmpwk) [inline planning/behavior_path_planner/src/scene_module/lane_change/util.cpp:132]: At L159, it is still `minimum_lane_change_length + buffer`. Is that correct?
```
      s_end = std::min(s_end, lane_length - num * (minimum_lane_change_length + buffer));
```

Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/src/scene_module/lane_change/util.cpp:132]: yes, that is correct.

Referring to the image below.
The L132 as you commented only compute the lane changing distance (only the green part)
While L159, computes purple and green part. 
image

Comment (shmpwk) [reply planning/behavior_path_planner/src/scene_module/lane_change/util.cpp:132]: Got it, thank you!!

**Review by rej55 — APPROVED** [review]
LGTM

**Pull request merged**